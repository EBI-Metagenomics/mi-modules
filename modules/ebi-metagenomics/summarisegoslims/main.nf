
process SUMMARISEGOSLIMS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.2.0--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:0.2.0--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(ips)
    tuple val(meta), path(gaf)
    path go_obo
    path go_banding

    output:
    tuple val(meta), path("*_summary.csv"), emit: go_summary
    tuple val(meta), path("*_slim.csv")   , emit: goslim_summary
    path "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    summarise_goslims \\
        -go ${go_obo} \\
        -gb ${go_banding} \\
        -i ${ips} \\
        -gaf ${gaf} \\
        -o ${prefix}_summary

    mv ${prefix}_summary ${prefix}_summary.csv
    mv ${prefix}_summary_slim ${prefix}_summary_slim.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_summary.csv
    touch ${prefix}_summary_slim.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """
}
