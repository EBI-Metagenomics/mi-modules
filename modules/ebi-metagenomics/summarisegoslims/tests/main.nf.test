
nextflow_process {

    name "Test Process SUMMARISEGOSLIMS"
    script "../main.nf"
    process "SUMMARISEGOSLIMS"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "summarisegoslims"

    test("summarisegoslims") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/generategaf/tests/ips_out.tsv", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/owltools/tests/ERRTESTING_ips_annotations.gaf", checkIfExists: true)
                ]
                input[2] = file("${baseDir}/modules/ebi-metagenomics/owltools/tests/go-dummy.obo", checkIfExists: true)
                input[3] = file("${moduleDir}/tests/goslim_banding_2024.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("summarisegoslims - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/generategaf/tests/ips_out.tsv", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/owltools/tests/ERRTESTING_ips_annotations.gaf", checkIfExists: true)
                ]
                input[2] = file("${baseDir}/modules/ebi-metagenomics/owltools/tests/go-dummy.obo", checkIfExists: true)
                input[3] = file("${moduleDir}/tests/goslim_banding_2024.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
