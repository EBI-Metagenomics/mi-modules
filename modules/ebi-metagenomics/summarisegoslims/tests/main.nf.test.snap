{
    "summarisegoslims": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary:md5,ddf9aa47da0a848795fffb56d9ac3c96"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim:md5,ce50e0507a12ef0d70bd41b14926c5d5"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c487c47425746717f82430c51a5688ec"
                ],
                "go_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary:md5,ddf9aa47da0a848795fffb56d9ac3c96"
                    ]
                ],
                "goslim_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim:md5,ce50e0507a12ef0d70bd41b14926c5d5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c487c47425746717f82430c51a5688ec"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T11:18:05.198568257"
    },
    "summarisegoslims - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c487c47425746717f82430c51a5688ec"
                ],
                "go_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "goslim_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c487c47425746717f82430c51a5688ec"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T11:18:11.207305173"
    }
}